enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java

```diff
@@ -100,43 +101,18 @@ public OnboardingServiceImpl(
      */
     @Transactional
     public OnboardingStartResponse startOnboarding(OnboardingStartRequest request) throws OnboardingProcessException, OnboardingOtpDeliveryException, TooManyProcessesException {
-        Map<String, Object> identification = request.getIdentification();
-        String identificationData = serializer.serialize(identification);
+        final Map<String, Object> identification = request.getIdentification();
+        final String identificationData = serializer.serialize(identification);
 
-        // Lookup user using identification attributes
-        String userId;
-        try {
-            final LookupUserRequest lookupUserRequest = LookupUserRequest.builder()
-                    .identification(identification)
-                    .build();
-            userId = onboardingProvider.lookupUser(lookupUserRequest).getUserId();
-        } catch (OnboardingProviderException e) {
-            logger.warn("User lookup failed, error: {}", e.getMessage(), e);
-            throw new OnboardingProcessException();
-        }
-
-        Optional<OnboardingProcessEntity> processOptional = onboardingProcessRepository.findExistingProcessForUser(userId, OnboardingStatus.ACTIVATION_IN_PROGRESS);
-        OnboardingProcessEntity process;
-        if (processOptional.isPresent()) {
-            // Resume an existing process
-            process = processOptional.get();
-            // Use latest identification data
-            process.setIdentificationData(identificationData);
-            process.setTimestampLastUpdated(new Date());
-        } else {
-            // Create an onboarding process
-            process = new OnboardingProcessEntity();
-            process.setIdentificationData(identificationData);
-            process.setStatus(OnboardingStatus.ACTIVATION_IN_PROGRESS);
-            process.setUserId(userId);
-            process.setTimestampCreated(new Date());
-        }
-        process = onboardingProcessRepository.save(process);
+        final OnboardingProcessEntity process = onboardingProcessRepository.findExistingProcessByIdentificationData(identificationData, OnboardingStatus.ACTIVATION_IN_PROGRESS)
```

Comment:
The change is covered by this new test case https://github.com/wultra/powerauth-tests/pull/86